paml/__init__.py

```diff
@@ -306,6 +306,84 @@ def mark_optional(parameter):
             """
 Primitive.__str__ = primitive_str
 
+
+def behavior_execution_parameter_value_map(self):
+    """
+    Return a dictionary mapping parameter names to value or (value, unit)
+    :param self:
+    :return:
+    """
+    parameter_value_map = {}
+
+    for pv in self.parameter_values:
+        name = pv.parameter.lookup().name
+        if isinstance(pv.value, uml.LiteralReference):
+            ref = pv.value.value.lookup()
+            value = ref.value if isinstance(ref, uml.LiteralSpecification) else ref
+            unit = ref.unit if isinstance(ref, uml.LiteralSpecification) and hasattr(ref, "unit") else None
+        else:
+            value = pv.value.value
+            unit = pv.value.unit if hasattr(pv.value, "unit") else None
+
+        parameter_value_map[name] = {"parameter" : pv.parameter.lookup(),
+                                     "value" : (value, unit) if unit else value}
+    return parameter_value_map
+BehaviorExecution.parameter_value_map = behavior_execution_parameter_value_map
+
+def call_behavior_execution_compute_output(self, parameter):
+    """
+    Get parameter value from call behavior execution
+    :param self:
+    :param parameter: output parameter to define value
+    :return: value
+    """
+    primitive = self.node.lookup().behavior.lookup()
+    call = self.call.lookup()
+    inputs = [x for x in call.parameter_values if x.parameter.lookup().direction == uml.PARAMETER_IN]
+    value = primitive.compute_output(inputs, parameter)
+    return value
+CallBehaviorExecution.compute_output = call_behavior_execution_compute_output
+
+def primitive_compute_output(self, inputs, parameter):
```

Comment:
I don't think this belongs in paml/__init__.py
In order to allow scaling, I would think that inference of outputs should be associated with the execution engine and with the individual primitives.
Doesn't need to change for the demo today, but shouldn't be kept this way for future development.